shadows/framework/src/main/java/org/robolectric/shadows/ShadowContextImpl.java

```diff
@@ -438,6 +440,21 @@ public File getDatabasePath(String name) {
     }
   }
 
+  @Implementation
+  protected SharedPreferences getSharedPreferences(String name, int mode) {
+    // Windows does not allow colons in file names, which may be used in shared preference
+    // names. URL-encode any colons in Windows.
+    if (!Strings.isNullOrEmpty(name) && File.separatorChar == '\\') {
+      name = name.replace(":", "%3A");
```

Comment:
To protect against naming conflicts, replace the escape character as well.

```java
name = name.replace("%", "%20").replace(":", "%3A")
```

Maybe we should just apply URL encoding altogether, e.g. using `java.net.URLEncoder`. The following characters are not allowed on NTFS volumes: `" * / : < > ? \ |`. URL encoding would replace each of those.

Please note that this change will break access to existing shared preferences (on linux). But I suppose Robolectric is not used that way.